src/content/get-started/logging.md

```diff
@@ -0,0 +1,404 @@
+Logging is a critical tool for debugging, monitoring, and understanding the
+behavior of your application in different environments.
+
+:::secondary What you'll learn
+
+*  Add the `logging` package to your project.
+*  Understand different log levels and how to use them.
+*  Create a `Logger` instance and configure it to write to a file.
+*  Log errors and warnings to a file for later inspection.
+
+:::
+
+## Prerequisites
+
+Before you begin this chapter, ensure you:
+
+*  Have completed Chapter 11 and have a working Dart development environment
+   with the `dartpedia` project.
+*  Understand the basics of debugging and why it's important to track errors and
+   events in your application.
+
+## Tasks
+
+In this chapter, you'll add logging to the `dartpedia` CLI application to help
+track errors and monitor its behavior. This will involve adding the `logging`
+package, creating a `Logger` instance, and writing log messages to a file.
+
+### Task 1: Add the `logging` package
+
+First, you need to add the `logging` package to your project's dependencies.
+
+1.  Open the `cli/pubspec.yaml` file.
+
+2.  Locate the `dependencies` section.
+
+3.  Add the `logging` package to your dependencies:
+
+    ```yaml
+    dependencies:
+      http: ^1.3.0
+      command_runner:
+        path: ../command_runner
+      wikipedia:
+        path: ../wikipedia
+      # Add the following line
+      logging: ^1.2.0
+    ```
+
+4.  Run `dart pub get` in the `cli` directory to fetch the new dependency.
+
+### Task 2: Create a logger
+
+Next, you'll create a `Logger` instance and configure it to write log messages
+to a file. This will involve creating a new file for the logger and setting up
+the necessary imports.
+
+1.  Create a new file called `cli/lib/src/logger.dart`.
+
+2.  Add the necessary imports to the top of the file:
+
+    ```dart title="cli/lib/src/logger.dart"
+    import 'dart:io';
+
+    import 'package:logging/logging.dart';
+    ```
+
+3.  Define a function called `initFileLogger` that configures and returns a
+    `Logger` instance.
+
+    ```dart title="cli/lib/src/logger.dart"
+    Logger initFileLogger(String name) {
+      hierarchicalLoggingEnabled = true;
+      final logger = Logger(name);
+      final now = DateTime.now();
+    
+      final segments = Platform.script.path.split('/');
+      final projectDir = segments.sublist(0, segments.length - 2).join('/');
+      final dir = Directory('$projectDir/logs');
+      if (!dir.existsSync()) dir.createSync();
+      final logFile = File(
+        '${dir.path}/${now.year}_${now.month}_${now.day}_$name.txt',
+      );
+    
+      logger.level = Level.ALL;
+      logger.onRecord.listen((record) {
+        final msg =
+            '[${record.time} - ${record.loggerName}] ${record.level.name}: ${record.message}';
+        logFile.writeAsStringSync('$msg \n', mode: FileMode.append);
+      });
+    
+      return logger;
+    }
+    ```
```

Comment:
This function is pretty long. It'd help to write in-line comments in each "section", where there are empty lines right now, or split it into 3 or 4  snippets that build up the function iteratively.